multiboot2/src/lib.rs

```diff
@@ -1262,7 +1265,7 @@ mod tests {
         assert_eq!(addr + bytes.len(), bi.end_address());
         assert_eq!(bytes.len(), bi.total_size());
         let es = bi.elf_sections_tag().unwrap();
-        let mut s = es.sections();
+        let mut s = es.sections(bi.offset);
```

Comment:
It's not entirely clear why it is now required to have this parameter in the public API and which are the tyical values that users want to pass in.. @YtvwlD

Reply:
if possible, please add more documentation with some guidance

Reply:
Oh, yes, this is because `BootInformation::elf_sections_tag` returns the real sections tag since 41428fbe57c84958bdbc4cce810e35bc6ac71975 and not a wrapper, so I had to(?) remove the passing of `self.offset` there.

Perhaps this should be `BootInformation::elf_sections` and just return the iterator instead of the tag?
The tag has no other public fields or methods.

As for the offset: I don't really know. This is the same offset that's passed to `BootInformation::load_with_offset`, `0` would be a typical value.

Reply:
I've done this in the latest commit, let me know what you think.